src/persist/src/indexed/mod.rs

```diff
@@ -334,8 +335,28 @@ impl<U: Buffer, L: Blob> Indexed<U, L> {
     /// smarts about waiting to call it only after there have been some writes),
     /// but it's exposed this way so we can write deterministic tests.
     pub fn step(&mut self) -> Result<(), Error> {
-        self.drain_future()
-        // TODO: Incrementally compact future.
+        self.drain_future()?;
+        let compaction_start = Instant::now();
+        let prev = self.serialize_meta();
+        for (_, future) in self.futures.iter_mut() {
+            if let Err(e) = future.step(&mut self.blob) {
+                self.restore(prev);
+                return Err(e);
+            }
+        }
+        for (_, trace) in self.traces.iter_mut() {
+            if let Err(e) = trace.step(&mut self.blob) {
+                self.restore(prev);
+                return Err(e);
+            }
+        }
+        self.try_set_meta(prev)?;
```

Comment:
found a bug! this should be `self.try_set_meta(self.serialize_meta())?;` :D

Reply:
fixed! the new setup in main where try_set_meta and restore do the correct thing by default and don't need the user to specify the restore point is much nicer!